books/calculus/bake

```diff
@@ -0,0 +1,197 @@
+
+    # This is somewhat backwards from how chemistry does it.  Just above we moved all of the review
+    # exercises to the end of the chapter.  Now, with all of the non-checkpoint exercises in the
+    # correct order, we bake them (the "in place" stuff) and number them.
+    # THIS IS HOW WE GET SECTION EXERCISES AND REVIEW EXERCISES NUMBERED IN ORDER
+    chapter.search('.section-exercises, .review-exercises').exercises.each do |exercise|
+      BakeNumberedExercise.v1(exercise: exercise, number: exercise.count_in(:chapter))
+    end
+
+    # chapter.notes('$.theorem').each do |theorem|
+    #   BakeTheorem.v1(theorem: theorem,
+    #                  number: "#{chapter.count_in(:book)}.#{theorem.count_in(:chapter)}")
+    # end
+
+    # chapter.notes('$.checkpoint').each do |checkpoint|
+    #   BakeCheckpoint.v1(checkpoint: checkpoint,
+    #                     number: "#{chapter.count_in(:book)}.#{checkpoint.count_in(:chapter)}")
+    # end
+  end
+
+  BakeChapterIntroductions.v1(book: book)
+  BakeChapterTitle.v1(book: book)
+
+  book.chapters.each do |chapter|
+    chapter.tables('$:not(.unnumbered)').each do |table|
+      BakeNumberedTable.v1(table: table,
+                           number: "#{chapter.count_in(:book)}.#{table.count_in(:chapter)}")
+    end
+
+    chapter.examples.each do |example|
+      BakeExample.v1(example: example,
+                     number: "#{chapter.count_in(:book)}.#{example.count_in(:chapter)}",
+                     title_tag: 'h3')
+    end
+
+    chapter.figures.each do |figure|
+      BakeFigure.v1(figure: figure,
+                    number: "#{chapter.count_in(:book)}.#{figure.count_in(:chapter)}")
+    end
+
+    BakeNonIntroductionPages.v1(chapter: chapter)
+  end
+
+  book.pages('$.appendix').each do |page|
+    appendix_letter = [*('A'..'Z')][page.count_in(:book) - 1]
+
+    page.figures.each do |figure|
+      BakeFigure.v1(figure: figure, number: "#{appendix_letter}#{figure.count_in(:page)}")
+    end
+
+    page.tables('$:not(.unnumbered)').each do |table|
+      BakeNumberedTable.v1(table: table, number: "#{appendix_letter}#{table.count_in(:page)}")
+    end
+
+    page.examples.each do |example|
+      BakeExample.v1(example: example,
+                     number: "#{appendix_letter}#{example.count_in(:page)}",
+                     title_tag: 'div')
+    end
+
+    BakeAppendix.v1(page: page, number: appendix_letter)
+  end
+
+  # Ok so here we go, now let's move the solutions to the end of the book.  As you know,
+  # unlike Chemistry, Calculus has an "Answer Key" composite chapter after the appendices.
+  # Similar to how we made a chapter review composite chapter above, let's make one
+  # for the Answer Key and then iterate over the chapters, making the chapter's answer key
+  # composite page which we append to the Answer Key composite chapter.
+
+  # New direction that makes the composite chapter Answer Key, appends it at the end of the
+  # book, then returns it
```

Comment:
Here too let's simplify the comment.  Change this if it doesn't contain enough info:

> Here we move the solutions to the end of the book.  Calculus has an "Answer Key" composite chapter after the appendices.  So we make the answer key, then iterate over the chapters, making an answer key composite page for each chapter that we append to the answer key composite chapter.